File: wxdgaming.boot.jscript/src/test/resources/SeedRandom.js
class SeedRandom {

    oldSeed = 0;
    addSeed = 0;

    constructor(seed) {
        this.oldSeed = seed;
    }

    refreshSeed() {
        this.oldSeed = (this.oldSeed * 9301 + this.addSeed++) & 233280;
    }

    nextInt(min, max) {
        if (max <= min) {
            return min;
        }
        let tmp = max - min;
        let next00 = this.next();
        return Math.round(min + Math.round(tmp * next00));
    }


    nextFloat(min, max) {
        if (max <= min) {
            return min;
        }
        let tmp = max - min;
        let next00 = this.next();
        return min + tmp * next00;
    }


    /**
     * 自定义随机算法, 0 ~ 1
     */
    next() {
        this.refreshSeed();
        let v = this.oldSeed / 233280.0;
        if (v < 0) {
            v = 0;
        }
        if (v > 1) {
            v = 1;
        }
        return v;
    }

}

function onInit() {
    console.log("js SeedRandom.onInit")
    let seedRandom = new SeedRandom(9527003456)
    for (let i = 1; i <= 10; i++) {
        console.log(i, seedRandom.nextInt(1, 1000))
    }
}

File: wxdgaming.boot.jscript/src/test/java/test/SeedRandom.java
package test;

/**
 * 种子随机数
 *
 * @author: wxd-gaming(無心道, 15388152619)
 * @version: 2025-02-27 10:18
 **/
public class SeedRandom {
    long oldSeed;
    long addSeed = 0;

    public SeedRandom(long oldSeed) {
        this.oldSeed = oldSeed;
    }

    /** 刷新随机数种子 */
    void refreshSeed() {
        oldSeed = (oldSeed * 9301 + addSeed++) & 233280L;
    }

    /** 随机范围 */
    public int nextInt(int min, int max) {
        if (max <= min) {
            return min;
        }
        int tmp = max - min;
        float next = next();
        return min + ((int) Math.floor(tmp * next));
    }

    /** float 下一个 */
    public float nextFloat(float min, float max) {
        if (max <= min) {
            return min;
        }
        float tmp = max - min;
        float next = next();
        return min + tmp * next;
    }

    /** 自定义随机算法, 0 ~ 1 */
    public float next() {
        refreshSeed();
        float v = oldSeed / 233280.0f;
        if (v < 0) {
            v = 0;
        }
        if (v > 1) {
            v = 1;
        }
        return v;
    }
}
